Raise the limit on the length of user names to the most utmp will
allow.  Based on a patch by Rik Brown.  RT#6773.

diff -up krb5-appl-1.0.1/bsd/krshd.c krb5-appl-1.0.1/bsd/krshd.c
--- krb5-appl-1.0.1/bsd/krshd.c	2010-09-09 18:33:22.969924000 -0400
+++ krb5-appl-1.0.1/bsd/krshd.c	2010-09-09 18:34:09.691923991 -0400
@@ -175,6 +175,13 @@ char copyright[] =
 
 
 
+#ifdef HAVE_UTMP_H
+#include <utmp.h>
+#endif
+
+#ifndef UT_NAMESIZE	/* linux defines it directly in <utmp.h> */
+#define	UT_NAMESIZE	sizeof(((struct utmp *)0)->ut_name)
+#endif
 
 #define MAXRETRIES 4
 
@@ -443,7 +450,7 @@ char    username[32] = "LOGNAME=";
 #include <tmpdir.h>
 char tmpdir[64] = "TMPDIR=";
 #else
-char	username[20] = "USER=";
+char	username[UT_NAMESIZE + 6] = "USER=";
 #endif
 
 char	homedir[64] = "HOME=";
@@ -493,7 +500,7 @@ int maxlogs;
 #define NCARGS 1024
 #endif
 
-#define NMAX   16 
+#define NMAX   UT_NAMESIZE
 
 int pid;
 char locuser[NMAX+1];
